go.mod

```diff
@@ -1,6 +1,7 @@
 module github.com/crunchydata/postgres-operator
 
-go 1.22.7
+// If this is changing when you don't want it to, see hack/go-get.sh
+go 1.23.0
```

Comment:
@benjaminjb I added a comment here, and it sticks around when using `go get`, `go mod tidy`, and `hack/go-get.sh`